Build is broken on MacOS

**Describe the bug**
PR #635, CL 14b6e7df63cf8f77bd3f4a493f9662bc9b5d77f4 broke MacOS build.

FAILED: src/vsg/CMakeFiles/vsg.dir/io/FileSystem.cpp.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DGLSLANG_EShLangRayGen -DGLSLANG_ResourceLimits_maxDualSourceDrawBuffersEXT -DGLSLANG_ResourceLimits_maxMeshViewCountEXT -I/Users/artemkulyk/VulkanSceneGraph/VulkanSceneGraph/include -I/Users/artemkulyk/VulkanSceneGraph/VulkanSceneGraph/cmake-build-debug/include -isystem /usr/local/include -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk -fPIC -fcolor-diagnostics -std=gnu++17 -MD -MT src/vsg/CMakeFiles/vsg.dir/io/FileSystem.cpp.o -MF src/vsg/CMakeFiles/vsg.dir/io/FileSystem.cpp.o.d -o src/vsg/CMakeFiles/vsg.dir/io/FileSystem.cpp.o -c /Users/artemkulyk/VulkanSceneGraph/VulkanSceneGraph/src/vsg/io/FileSystem.cpp
/Users/artemkulyk/VulkanSceneGraph/VulkanSceneGraph/src/vsg/io/FileSystem.cpp:120:19: error: variable has incomplete type 'struct stat64'
    struct stat64 stbuf;
                  ^
/Users/artemkulyk/VulkanSceneGraph/VulkanSceneGraph/src/vsg/io/FileSystem.cpp:120:12: note: forward declaration of 'stat64'
    struct stat64 stbuf;
           ^
1 error generated.

**To Reproduce**
Steps to reproduce the behavior:
cmake && make on MacOS

**Expected behavior**
Successful build

**Desktop (please complete the following information):**
- OS: Mac OS Ventura
- HW: Apple Silicon (M1)
- Compiler: Apple clang version 14.0.0 (clang-1400.0.29.202)
- Target: arm64-apple-darwin22.1.0
- MoltenVK Vulkan SDK 1.3.231
- VulkanSceneGraph master

**Additional context**
See build logs below.
[build.log](https://github.com/vsg-dev/VulkanSceneGraph/files/10189091/build.log)

--- a/src/vsg/io/FileSystem.cpp
+++ b/src/vsg/io/FileSystem.cpp
@@ -116,6 +116,9 @@ FileType vsg::fileType(const Path& path)
 #if defined(_MSC_VER)
     struct __stat64 stbuf;
     if (_wstat64(path.c_str(), &stbuf) != 0) return FILE_NOT_FOUND;
+#elif defined(__APPLE__)
+    struct stat stbuf;
+    if (stat(path.c_str(), &stbuf) != 0) return FILE_NOT_FOUND;
 #else
     struct stat64 stbuf;
     if (stat64(path.c_str(), &stbuf) != 0 ) return FILE_NOT_FOUND;

